### PR TITLE
Add release process

### PR DESCRIPTION
**Pull request by clouserw**

### CODE REVIEW

File: docs/engineering/dev-process.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    0. Notes from the deploy doc, particularly any server side changes that
+       need to happen as part of this deployment.
+    0. Links to the needs:qa labels on GitHub.
+    0. Links to the release tags on GitHub.
+    0. Links to pertinent changelogs.
+    0. Links to Circle builds that push to DockerHub.
+
+#### Operations staff will take it from there…
+0. Ensure that any configuration changes noted in the deployment bug land in [cloudops-deployment][cloudops-deployment].
+0. Run any outstanding database migrations
```
Comment (clouserw) [inline docs/engineering/dev-process.md]: Where do these come from?  Do they run automatically for dev/stage?

File: docs/engineering/dev-process.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+0. Build fxa-auth, fxa-content, fxa-oauth, fxa-profile, fxa-verifier in stage on Jenkins using git commit from cloudops-deployment PR and docker images referenced in deploy bug
+0. Ensure that TeamCity tests are passing
+0. Request QA on stage
+    0. If major issues are found, a new patch is made and we’re back to step 3, running `release.sh patch`
+        0. This command assumes that the relevant commit has been merged into the appropriate train branch. It bumps the minor rev on the last tag it finds in the tree from HEAD, so also requires that you checkout the appropriate train branch locally.
+0. Deploy fxa-* to prod
```
Comment (clouserw) [inline docs/engineering/dev-process.md]: Do we have any QA once we deploy to prod?

File: docs/engineering/dev-process.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+0. Initial deployment bug is closed
+
+
+
+FAQ:
+* How to spin up a new dev/stage environment?
```
Comment (clouserw) [inline docs/engineering/dev-process.md]: Maybe there is documentation for this somewhere?

File: docs/engineering/dev-process.md (hunk 1 above)
Comment (clouserw) [reply docs/engineering/dev-process.md]: They are in the repositories.  They are applied automatically to dev & stage.

File: docs/engineering/dev-process.md (hunk 2 above)
Comment (clouserw) [reply docs/engineering/dev-process.md]: Not really

File: docs/engineering/dev-process.md (hunk 3 above)
Comment (clouserw) [reply docs/engineering/dev-process.md]: stage is hard.  fxa-dev repository is easier.  The readme for that is the documentation

**Comment by rfk** [comment]
I figure we'll probably have a few conversations about whats in or out of scope for this repo, so I may as well start one now...this doesn't feel like it belongs in this repo to me.

These are internal team docs for use by engineers working on FxA. They're very important! But they don't need to be on an outward-facing docs site to be discoverable by the broader Mozilla org, and they risk making it harder for that target audience to find the content they actually need. It feels like something that should live in https://github.com/mozilla/fxa rather than here.

(Feel free to disagree with this of course! It's just my gut reaction based on what I think this repo is trying to accomplish).

**Comment by clouserw** [comment]
You might be right!  At this point I'm trying to collect documentation that is scattered around in google docs and gists and put it in one place.  Once it's collected it will be easier to see an organizational pattern and we can put things in the best places.  If something shouldn't be here (eg. I don't plan on moving the ADUs) I expect to link to it from here, at least.

**Comment by rfk** [comment]
> I expect to link to it from here, at least.

👍 I think "everything should be discoverable starting from here" is an excellent ambition

File: docs/engineering/dev-process.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+## Releasing Code
+0. A release owner is delegated to follow the process below and work with the
+   team to tie up any loose ends
+0. The pre-flight checklist:
+    0. Ensure there are no critical patches for this tag that haven't landed
+       yet
+    0. Fill out a new section in the [deployment doc][deployment-doc] for the Train you are tagging
+    0. Ensure you've got appropriate qa signoffs (**todo: link to doc**)
+    0. Ensure you don't have any modified files or code laying around before
+       you start the tag (**todo: does it matter what your github origins
```
Comment (clouserw) [inline docs/engineering/dev-process.md]: @vbudhram do you know this answer?

Comment (vbudhram) [reply docs/engineering/dev-process.md]: No, it doesn't. The release script will automatically create the private ones for you, https://github.com/mozilla/fxa/blob/master/release.sh#L324.

File: docs/engineering/dev-process.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+0. A release owner is delegated to follow the process below and work with the
+   team to tie up any loose ends
+0. The pre-flight checklist:
+    0. Ensure there are no critical patches for this tag that haven't landed
+       yet
+    0. Fill out a new section in the [deployment doc][deployment-doc] for the Train you are tagging
+    0. Ensure you've got appropriate qa signoffs
+    0. Ensure you don't have any modified files or code laying around before
+       you start the tag
+0. Run [release.sh][release.sh] from the root of the repository.  Make sure
```
Comment (vbudhram) [inline docs/engineering/dev-process.md]: Not sure if it makes sense to mention point releases here (maybe that is worth its own section). Basically it is same process except you run with `release.sh patch`, while in the train branch.

File: docs/engineering/dev-process.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+   there are no errors in the output.
+0. Do some manual checks to make sure the generated tags are sane:
+    0. Do the changelogs match expectations from `git log`?
+    0. Have all the version strings been updated?
+    0. Does the diff from `origin/master` (or `origin/train-xxx` if it’s a point release) look correct?
+    0. Does the diff between the public and private tags look correct?
+0. The release script will print some commands to run to push the public and private train branches to the remotes.  **It's best to copy and paste these so you don't mix them up.**
+0. The release script will also print some URLs which you can use to open PRs to merge the train branches back to their respective master branches
+0. Open a deployment bug in bugzilla under `Cloud Services :: Operations: Deployment Requests` ([example][example-deployment-bug]). Remember to include:
```
[the remark this answers is not in the document]
Comment (clouserw) [reply docs/engineering/dev-process.md]: done thanks

File: docs/engineering/dev-process.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+---
+id: dev-process
+title: Development Process
+sidebar_label: Development Process
+---
+
+
+
+## Releasing Code
+0. A release owner is delegated to follow the process below and work with the
```
[the remark this answers is not in the document]
Comment (clouserw) [reply docs/engineering/dev-process.md]: done

File: docs/engineering/dev-process.md (hunk 5 above)
Comment (clouserw) [reply docs/engineering/dev-process.md]: I added an FAQ section at the bottom, although if you want to expand, please do

**Pull request merged**